pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -95,6 +95,11 @@ def reset(self) -> None:
         assert self.batch_loop.optimizer_loop is not None
         self.is_last_batch = False
 
+        if self.restarting:
```

Comment:
Shouldn't it be `not`?

```suggestion
        if not self.restarting:
```

Reply:
This is correct.